presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -433,7 +437,7 @@ public PlanOptimizers(
                     ruleStats,
                     statsCalculator,
                     estimatedExchangesCostCalculator,
-                    ImmutableSet.of(new DetermineJoinDistributionType(costComparator))))); // Must run before AddExchanges
+                    ImmutableSet.of(new DetermineJoinDistributionType(costComparator, exchangeCostCalculator))))); // Must run before AddExchanges
```

Comment:
Cost calculators are passed to Rules from IterativeOptimizer, via the Context

Reply:
Based on the offline discussion with @rschlussel it was decided that the `ExchangeCostCalculator` is a bad abstraction. Instead we introduced the `TaskCountEstimator`.